src/zenml/orchestrators/step_launcher.py

```diff
@@ -209,7 +210,9 @@ def launch(self) -> None:
                 )
                 try:
                     execution_needed, step_run = self._prepare(
-                        step_run=step_run
+                        step_run=step_run,
+                        model_config=step_run.config.model_config
+                        or pipeline_run.config.model_config,
```

Comment:
Same as in the other PR - these will no longer exist in the step_run / pipeline_run models after #1835

Reply:
Instead, you can get the configs from the deployment inside the `_prepare` function directly via `self._deployment`

Reply:
Reworked